### PR TITLE
feat(backup): Support org slug remapping

### LINKED ISSUE #170: Add toggle for "INSERT-only" vs "INSERT_OR_UPDATE" modes on import

### LINKED ISSUE #182: Support org slug mapping

### PR DESCRIPTION
**Pull request by azaslavsky**
We now handle organization slug collisions gracefully by extending the slug with a random suffix (we do this for new organizations already). Comparators have also been extended to recognize and pass auto-suffixed slugs of this kind.

Closes getsentry/team-ospo#170
Closes getsentry/team-ospo#182

### CODE REVIEW

File: src/sentry/api/endpoints/user_details.py [hunk 1]
```diff
@@ -75,7 +75,11 @@ class UserOptionsSerializer(serializers.Serializer):
 
 class BaseUserSerializer(CamelSnakeModelSerializer):
     def validate_username(self, value):
-        if User.objects.filter(username__iexact=value).exclude(id=self.instance.id).exists():
+        if (
+            User.objects.filter(username__iexact=value)
+            .exclude(id=self.instance.id if hasattr(self.instance, "id") else 0)
```
Comment (hubertdeng123) [inline src/sentry/api/endpoints/user_details.py:80]: What's the reasoning behind defaulting to 0, is that the supposed default instance id or something?

File: src/sentry/models/userip.py [hunk 2]
```diff
@@ -40,6 +44,44 @@ def log(cls, user: User | RpcUser, ip_address: str):
...
+            return None
+
+        # We'll recalculate the country codes from the IP when we call `log()` in
+        # `write_relocation_import()`.
+        self.country_code = None
+        self.region_code = None
+
+        # Only preserve the submitted timing data in the global scope.
+        if scope != ImportScope.Global:
+            self.first_seen = timezone.now()
```
Comment (hubertdeng123) [inline src/sentry/models/userip.py:61]: nit
```suggestion
            self.first_seen = self.last_seen = timezone.now()
```

File: src/sentry/api/endpoints/user_details.py (hunk 1 above)
Comment (azaslavsky) [reply src/sentry/api/endpoints/user_details.py]: 0 can't be a pk (1-indexed). Seemed simpler than writing a big branching logic block.

File: src/sentry/models/userip.py (hunk 2 above)
Comment (azaslavsky) [reply src/sentry/models/userip.py]: Done

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**